dynamiqs/utils/utils.py

```diff
@@ -603,14 +659,14 @@ def fidelity(x: Tensor, y: Tensor) -> Tensor:
         >>> fock0 = dq.fock(3, 0)
         >>> dq.fidelity(fock0, fock0)
         tensor(1.)
-        >>> fock01 = 0.5 * (dq.fock_dm(3, 1) + dq.fock_dm(3, 0))
-        >>> dq.fidelity(fock01, fock01)
+        >>> fock01_dm = 0.5 * (dq.fock_dm(3, 1) + dq.fock_dm(3, 0))
+        >>> dq.fidelity(fock01_dm, fock01_dm)
         tensor(1.000)
-        >>> dq.fidelity(fock0, fock01)
+        >>> dq.fidelity(fock0, fock01_dm)
         tensor(0.500)
     """
     if isket(x) and isket(y):
-        return _ket_fidelity(x, y)
+        return overlap(x, y)
```

Comment:
I think the fidelity also boils down to the overlap if one of the two argument is a ket.